### ISSUE TITLE
Component for FIs on home page

### ISSUE THREAD

**Issue by awolfe76** [cfpb/hmda-platform-ui#30]
This component will render an available FI for a user to begin, continue, or complete a filing.

Each item in the list will contain:
- FI name
- ~~a link for the user to verify the FI information~~ TBD
- the date/time the filing started, if it has started
- a description of the current progress and relevant links
- list of available reports (edit, summary, etc), if they exist

---
### An example of an institution that has not started filing:

![fi-not-started-example](https://cloud.githubusercontent.com/assets/2932572/13464140/ba416f56-e05d-11e5-8181-da777dfd2f25.png)

---
### Several more examples of different states:

![fi-many-examples](https://cloud.githubusercontent.com/assets/2932572/13464122/9428da66-e05d-11e5-9868-740e577d676b.png)

---
### An example of a completed state:

![fi-complete-example](https://cloud.githubusercontent.com/assets/2932572/13464177/dfb9f186-e05d-11e5-8aeb-d1aa26bfd643.png)

**Comment by awolfe76** [comment]
In order to render an institution (or several institutions) and its state an API call might look like:

`/institutions?user=[username]`

And might return a response of:

``` json
{
    "institutions": [
        {
          "name": "Bank 1",
          "status": {
                          "overall": 1,
              "syntactical": 2,
              "validity": 3,
              "quality": 2,
              "macro": 1
          },
          "date-started": "2017-02-23T18:25:43.511Z",
          "history": ["2017-02-23T18:25:43", "2017-02-22T11:13:33", "2017-02-21T09:43:08"]
        }, {
          "name": "Bank 2",
          "status": {
                          "overall": 4,
              "syntactical": 3,
              "validity": 3,
              "quality": 3,
              "macro": 3
          },
          "date-started": "2017-02-23T18:25:43.511Z",
          "history": ["2017-02-23T18:25:43", "2017-02-22T11:13:33", "2017-02-21T09:43:08"]
        }, {
          "name": "Bank n",
          "status": {
                          "overall": 1,
              "syntactical": 1,
              "validity": 1,
              "quality": 0,
              "macro": 0
          },
          "date-started": "2017-02-23T18:25:43.511Z",
          "history": ["2017-02-23T18:25:43", "2017-02-22T11:13:33", "2017-02-21T09:43:08"]
        }
    ]
}
```

where
- `date-started` allows us to render the `Filing started 22 minutes ago` or `Filing started 2 days ago`
- `history` allows us to render the list of edit reports (if we are providing them)
- `status` allows us to render the correct content and links
  - status codes could equal
    - 0 = hasn't started
    - 1 = running
    - 2 = done, with errors
    - 3 = done, no errors
    - 4 = signed

**Comment by wpears** [comment]
I was thinking about the "verify info" link... what if it is just presented every time the user tries to initially file for an institution each year? So before the upload screen, you get a verify information screen with what we have on file prepopulated.

We'd still need a way to edit this information on the fly, but it could be a separate link on the homepage that doesn't have to exist on every institution heading. Moreover, by presenting the "verify info" screen automatically, I think we'll help people catch more errors than if they have to click into a link when they think their info is already okay (when it might not be).

**Comment by lesgou** [comment]
@Kibrael and @lesgou are working on a workflow related to this during this sprint.

**Comment by awolfe76** [comment]
I agree, probably a better way to handle this.

> We'd still need a way to edit this information on the fly

Maybe starting to head down a "profile" type of page?

Interested to know if we have any idea how often the info is wrong. Might lead us in different directions.

Maybe the first year we do what @wpears recommends by showing them the info prior to the file upload. Then, based on usage, we can go another route?

Have fun @lesgou and @Kibrael!

**Comment by awolfe76** [comment]
Possible update to the API response. I noticed when creating #36 that the responses were very similar. So ...

``` json
{
  "institutions": [
    {
      "name": "Bank 1",
      "id": 1234567890,
      "date-started": "2017-02-23T18:25:43.511Z",
      "history": ["2017-02-23T18:25:43", "2017-02-22T11:13:33", "2017-02-21T09:43:08"],
      "status": 2,
      "rowsChecked": 33,
      "macroChecked": 0,
      "errors": {
        "syntactical": 2,
        "validity": 0,
        "quality": 1,
        "macro": 0
      }
    },
    {
      "name": "Bank n",
      "id": 0987654321,
      "date-started": "2017-02-22T18:25:43.511Z",
      "history": ["2017-02-22T14:51:58", "2017-02-21T11:03:48", "2017-02-20T11:17:32"],
      "status": 3,
      "rowsChecked": 33,
      "macroChecked": 15,
      "errors": {
        "syntactical": 10,
        "validity": 3,
        "quality": 5,
        "macro": 2
      }
    }
  ]  
}
```

This allows us to render everything on the home page, plus pass this data to the progress screen in the case where a filer clicks an institution that has started filing. We will still need the API call in #36 to handle updating the UI in that case, but this might prevent some duplication on the backend.

Plus with the possible `status` values:
- 0 = uploading
- 1 = parsing
- 2 = row checks (syntax, validity, quality)
- 3 = macro
- 4 = checks complete
- 5 = signed

we would know on the progress UI to not start polling when `status > 3`.

@jmarin

**Comment by wpears** [comment]
Closed by #52
